src/Persistence/Proxy/PersistedObjectsTracker.php

```diff
@@ -21,9 +21,14 @@ final class PersistedObjectsTracker
     /**
      * This buffer of objects needs to be static to be kept between two kernel.reset events.
      *
-     * @var list<\WeakReference<object>>
+     * @var \WeakMap<object, bool>
      */
-    private static $buffer = [];
+    private static \WeakMap $buffer;
```

Comment:
finally I'm using a weak map 😄

this prevents to have duplicates in the buffer